internal/services/api/post_deployment.go

```diff
@@ -76,16 +78,17 @@ func PostDeploymentHandlerFunc(
 		w.WriteHeader(http.StatusAccepted)
 		json.NewEncoder(w).Encode(response)
 
+		log := log.WithArgs("local_id", localID)
 		newState.LocalID = localID
 		publisher, err := publisherFactory(newState, emitter, log)
+		log.Debug("New publisher derived from state", "account", b.AccountName, "config", b.ConfigName)
 		if err != nil {
 			InternalError(w, req, log, err)
 			return
 		}
 
 		go func() {
-			log := log.WithArgs("local_id", localID)
```

Comment:
By removing this, are we losing functionality? How will we see the localID, which allows us to discriminate between logs of multiple streams?

Reply:
It is not being removed, it was moved to line 81 above so the derived logger with the `local_id` can be used earlier and passed down to the `publisherFactory`